pynumaflow/sourcer/_dtypes.py

```diff
@@ -171,6 +179,41 @@ def count(self) -> int:
         return self._count
 
 
+@dataclass(init=False)
+class PartitionsResponse:
+    """
+    PartitionsResponse is the response for the partition request.
+    It indicates the number of partitions at the user defined source.
+    A negative count indicates that the partition information is not available.
+    Args:
+        count: the number of partitions.
+    """
+
+    __slots__ = ("_partitions",)
+    _partitions: list[int]
+
+    def __init__(self, partitions: list[int]):
+        if not isinstance(partitions, list):
+            raise TypeError(f"Wrong data type: {type(partitions)} for Partition.partitions")
+        self._partitions = partitions
```

Comment:
```suggestion
    def __init__(self, *partitions: int):
        self._partitions = list(partitions)
```
And invoke like:
`PartitionsResponse(1, 2, 3)`
instead of:
`PartitionsResponse([1, 2, 3])`

You can also do it this way. But both are correct. Your choice. :)

Reply:
Have not changed this one right now! Will update in another iteration if needed.